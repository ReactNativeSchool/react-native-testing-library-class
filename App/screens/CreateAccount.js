import React from "react";
import {
  ScrollView,
  Text,
  TouchableOpacity,
  StyleSheet,
  View,
} from "react-native";

import { TextField, ErrorText } from "../components/Form";
import { Button } from "../components/Button";

const styles = StyleSheet.create({
  textBlock: {
    marginTop: 20,
  },
  text: {
    fontSize: 18,
    color: "#969696",
    textAlign: "center",
    marginBottom: 2,
  },
  link: {
    textDecorationLine: "underline",
  },
});

const isValidInputs = (state) => {
  const fields = ["email", "fName", "lName", "password", "cPassword"];
  const validArray = fields.map((field) => {
    if (!state[field] || state[field].length === 0) {
      return false;
    }
    return true;
  });

  const validFields = validArray.filter((valid) => valid);
  return validFields.length === fields.length;
};

export default class CreateAccount extends React.Component {
  state = { errorMessage: null };

  onSubmit = () => {
    if (!isValidInputs(this.state)) {
      this.setState({ errorMessage: "An error occured." });
    } else {
      this.setState({ errorMessage: null });
      const { email, fName, lName, password } = this.state;

      fetch("https://postman-echo.com/post", {
        method: "POST",
        headers: {
          "Content-Type": "application/json",
        },
        body: JSON.stringify({
          email,
          fName,
          lName,
          password,
        }),
      })
        .then((res) => res.json())
        .then((res) => {
          console.log("res", res);
        })
        .catch((err) => {
          console.log("err", err);
        });
    }
  };

  render() {
    const { navigation } = this.props;

    return (
      <ScrollView
        contentContainerStyle={{ paddingVertical: 20 }}
        style={{ backgroundColor: "#fff" }}
      >
        <TextField
          testID="CreateAccount.email"
          label="Email"
          placeholder="john.doe@example.com"
          onChangeText={(email) => this.setState({ email })}
        />
        <TextField
          label="First Name"
          placeholder="John"
          onChangeText={(fName) => this.setState({ fName })}
        />
        <TextField
          label="Last Name"
          placeholder="Doe"
          onChangeText={(lName) => this.setState({ lName })}
        />
        <TextField
          label="Password"
          secureTextEntry
          onChangeText={(password) => this.setState({ password })}
        />
        <TextField
          label="Confirm Password"
          secureTextEntry
          onChangeText={(cPassword) => this.setState({ cPassword })}
        />
        <ErrorText
          testID="CreateAccount.errorMessage"
          text={this.state.errorMessage}
        />
        <Button text="Submit" onPress={this.onSubmit} />
        <View style={styles.textBlock}>
          <Text style={styles.text}>Already have an account?</Text>
          <TouchableOpacity onPress={() => navigation.navigate("SignIn")}>
            <Text style={[styles.text, styles.link]}>Sign in.</Text>
          </TouchableOpacity>
        </View>
      </ScrollView>
    );
  }
}
