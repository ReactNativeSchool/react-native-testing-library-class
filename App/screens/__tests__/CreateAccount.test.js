import React from "react";
import { render } from "@testing-library/react-native";

import CreateAccount from "../CreateAccount";

test("it renders all inputs as expected", () => {
  const { toJSON } = render(<CreateAccount />);

  expect(toJSON()).toMatchSnapshot();
});
