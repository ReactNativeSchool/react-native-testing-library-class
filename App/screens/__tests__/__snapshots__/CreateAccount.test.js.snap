// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders all inputs as expected 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "paddingVertical": 20,
    }
  }
  style={
    Object {
      "backgroundColor": "#fff",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "borderBottomColor": "#E4E4E4",
          "borderBottomWidth": 1,
          "marginBottom": 11,
          "marginHorizontal": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#4A4A4A",
            "fontSize": 18,
            "fontWeight": "600",
            "marginBottom": 7,
          }
        }
      >
        Email
      </Text>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="john.doe@example.com"
        placeholderTextColor="#828282"
        rejectResponderTermination={true}
        style={
          Object {
            "color": "#828282",
            "fontSize": 18,
            "fontWeight": "400",
            "marginBottom": 4,
          }
        }
        testID="Form.TextInput"
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "#E4E4E4",
          "borderBottomWidth": 1,
          "marginBottom": 11,
          "marginHorizontal": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#4A4A4A",
            "fontSize": 18,
            "fontWeight": "600",
            "marginBottom": 7,
          }
        }
      >
        First Name
      </Text>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="John"
        placeholderTextColor="#828282"
        rejectResponderTermination={true}
        style={
          Object {
            "color": "#828282",
            "fontSize": 18,
            "fontWeight": "400",
            "marginBottom": 4,
          }
        }
        testID="Form.TextInput"
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "#E4E4E4",
          "borderBottomWidth": 1,
          "marginBottom": 11,
          "marginHorizontal": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#4A4A4A",
            "fontSize": 18,
            "fontWeight": "600",
            "marginBottom": 7,
          }
        }
      >
        Last Name
      </Text>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Doe"
        placeholderTextColor="#828282"
        rejectResponderTermination={true}
        style={
          Object {
            "color": "#828282",
            "fontSize": 18,
            "fontWeight": "400",
            "marginBottom": 4,
          }
        }
        testID="Form.TextInput"
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "#E4E4E4",
          "borderBottomWidth": 1,
          "marginBottom": 11,
          "marginHorizontal": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#4A4A4A",
            "fontSize": 18,
            "fontWeight": "600",
            "marginBottom": 7,
          }
        }
      >
        Password
      </Text>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholderTextColor="#828282"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "color": "#828282",
            "fontSize": 18,
            "fontWeight": "400",
            "marginBottom": 4,
          }
        }
        testID="Form.TextInput"
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "#E4E4E4",
          "borderBottomWidth": 1,
          "marginBottom": 11,
          "marginHorizontal": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#4A4A4A",
            "fontSize": 18,
            "fontWeight": "600",
            "marginBottom": 7,
          }
        }
      >
        Confirm Password
      </Text>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholderTextColor="#828282"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "color": "#828282",
            "fontSize": 18,
            "fontWeight": "400",
            "marginBottom": 4,
          }
        }
        testID="Form.TextInput"
        underlineColorAndroid="transparent"
      />
    </View>
    <Text
      style={
        Object {
          "color": "red",
          "fontSize": 16,
          "marginBottom": 15,
          "marginHorizontal": 20,
          "marginTop": 5,
        }
      }
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#D22322",
          "marginHorizontal": 20,
          "opacity": 1,
          "paddingHorizontal": 45,
          "paddingVertical": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 18,
            "fontWeight": "500",
          }
        }
      >
        Submit
      </Text>
    </View>
    <View
      style={
        Object {
          "marginTop": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#969696",
            "fontSize": 18,
            "marginBottom": 2,
            "textAlign": "center",
          }
        }
      >
        Already have an account?
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#969696",
                "fontSize": 18,
                "marginBottom": 2,
                "textAlign": "center",
              },
              Object {
                "textDecorationLine": "underline",
              },
            ]
          }
        >
          Sign in.
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
